src/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs

```diff
@@ -99,12 +99,18 @@ private static void AddPrefixInternal(string p, HttpListener listener)
                 throw new HttpListenerException((int)HttpStatusCode.BadRequest, SR.net_invalid_path);
 
             // listens on all the interfaces if host name cannot be parsed by IPAddress.
-            HttpEndPointListener epl = GetEPListener(lp.Host, lp.Port, listener, lp.Secure);
+            HttpEndPointListener epl = GetEPListener(lp.Host, lp.Port, listener, lp.Secure, out bool alreadyExists);
+            if (alreadyExists)
+            {
+                throw new HttpListenerException(98, SR.Format(SR.net_listener_already, p));
+            }
             epl.AddPrefix(lp, listener);
         }
 
-        private static HttpEndPointListener GetEPListener(string host, int port, HttpListener listener, bool secure)
+        private static HttpEndPointListener GetEPListener(string host, int port, HttpListener listener, bool secure, out bool alreadyExists)
         {
+            alreadyExists = false;
```

Comment:
Rather than doing this here, could it be done in the else block of the same if where you set it to true?